### PR TITLE
collect custom lint rules from internal dependencies

### PR DESCRIPTION
**Pull request by mohammadkahelghi-grabtaxi**
collect custom lint rules from internal dependencies 
with these changes we can use an Interanl target (java/kotlin) to write custom lint tools and check the code against it

### CODE REVIEW

**Review by roscrazy — APPROVED** [review]

File: rules/android/android_binary.bzl [hunk 1]
```diff
@@ -82,6 +82,7 @@ def android_binary(
             baseline = lint_baseline,
             lint_config = lint_options.get("config", None),
             deps = kotlin_library_deps,
+            custom_lint_rule_targets = lint_options.get("customLintTargets", default = []),
```
Comment (arunkumar9t2) [inline rules/android/android_binary.bzl:85]: Can we name this `lint_checks` to be consistent with what we are passing to `project.xml` instead? Both here and in `lint_options`?

Also should we apply this to `android_library` too?

**Comment by arunkumar9t2** [comment]
It is not clear how this will be used. Can we add a working sample and possibly update the baseline so we don't regress?

File: rules/android/android_binary.bzl (hunk 1 above)
Comment (mohammadkahelghi-grabtaxi) [reply rules/android/android_binary.bzl:85]: yes and yes

File: rules/android/lint/lint_sources.bzl [hunk 2]
```diff
@@ -64,6 +65,11 @@ lint_sources = rule(
             default = [],
             aspects = [collect_aar_aspect],
         ),
+        "lint_checks": attr.label_list(
```
Comment (arunkumar9t2) [inline rules/android/lint/lint_sources.bzl:68]: It is worth restricting the type of labels that can be supplied here. Leaving this empty might cause wrong jars to be passed unnecessarily. Preferably we can have a custom provider and allow targets only if they provide it.

**Review by arunkumar9t2 — APPROVED** [review]

**Pull request merged**